runbotics/runbotics-scheduler/src/scheduler-database/credential-collection/credential-collection.service.ts

```diff
@@ -0,0 +1,259 @@
+                credentialCollectionUser: {
+                    userId: user.id,
+                    privilegeType: PrivilegeType.WRITE,
+                },
+            },
+        });
+
+        if (collections.length === 0) {
+            throw new NotFoundException('Could not find any credential collections');
+        }
+
+        return collections;
+    }
+
+    async findOneAccessibleById(id: string, user: IUser) {
+        const collection = await this.credentialCollectionRepository.findOne({
+            where: {
+                id,
+                tenantId: user.tenantId,
+            },
+            relations,
+        });
+
+        if (!collection) {
+            throw new NotFoundException('Could not find credential collection wit id: ' + id);
+        }
+
+        if (
+            collection.credentialCollectionUser.some((item) => item.userId === user.id && item.privilegeType === PrivilegeType.WRITE)
+        ) {
+            throw new ForbiddenException('You do not have access to this collection');
+        }
+
+        return collection;
+    }
+
+    async findOneByCriteria(tenantId: string, criteria: Partial<CredentialCollection>) {
+        const collection = await this.credentialCollectionRepository.findOne({
+            where: {
+                tenantId,
+                ...criteria,
+            },
+            relations,
+        });
+
+        if (!collection) {
+            throw new NotFoundException('Could not find credential collection with given criteria');
+        }
+
+        return collection;
+    }
+
+    async update(
+        id: string,
+        updateCredentialCollectionDto: UpdateCredentialCollectionDto,
+        user: IUser,
+    ) {
+        const { sharedWith, ...dto } = updateCredentialCollectionDto;
+        const { tenantId } = user;
+
+        if (dto.accessType !== AccessType.GROUP) {
+            return this.credentialCollectionRepository.update(id, dto);
+        }
+
+        const credentialCollection =
+            await this.credentialCollectionRepository.findOne({
+                where: { id, tenantId },
+                relations,
+            });
+
+        if (!credentialCollection) {
+            throw new NotFoundException(`Could not find collection with id ${id}`);
+        }
+
+        const collectionUserArray = await this.credentialCollectionUserRepository.find({
+            where: {
+                credentialCollectionId: id,
+            },
+        });
+
+        const updatedCollectionUserArray =
+            await this.getCollectionUserArrayWithPrivileges(
+                sharedWith,
+                credentialCollection,
+                user,
+                tenantId
+            );
+
+        await collectionUserArray.forEach((item) => {
+            this.credentialCollectionUserRepository.remove(item);
+        });
+
+        await updatedCollectionUserArray.forEach((item) => {
+            this.credentialCollectionUserRepository.save(item);
+        });
```

Comment:
shouldn't you have used here:

await Promise.all(collectionUserArray.map((item) => {
            this.credentialCollectionUserRepository.remove(item);
        }))